modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala

```diff
@@ -156,7 +155,7 @@ object GlobalSnapshotConsensusFunctions {
       } yield (globalSnapshot, returnedEvents)
     }
 
-    private def updateBalancesByTransactions(balances: SortedMap[Address, Balance], txs: SortedSet[RewardTransaction]) =
+    private def updateBalancesByRewards(balances: SortedMap[Address, Balance], txs: SortedSet[RewardTransaction]) =
```

Comment:
Please take a look at that logic.
If someone overflows `NonNegLong` then his balance will be replaces with `tx.amount`. It is not the correct behavior but such case should never happen because the total supply is lower than `NonNegLong.MaxValue`. I can throw there if you guys want.

Reply:
Wait, I think you pointed out an error here. I believe it should be:

```scala
maybeBalance
  .map(balance => balance.plus(tx.amount).getOrElse(balance))
  .getOrElse(Balance.fromAmount(tx.amount))
```

Reply:
Fallback to previous balance means that reward transition exists but it didn't affect the balance :< so also incorrect but the lesser of 2 evils

Reply:
Currently, the logic is the following: if someone has a very high balance, i.e. `Long.MaxValue` and we want to reward him with `1L`, then his balance will be set to `1L`. This is incorrect as the balance should not decrease when adding rewards.

Regarding burning vs redistributing, I think we've discussed it already and the outcome was that burning is a safer option than redistributing.

Reply:
I mean the current logic is incorrect (that's why I started the discussion) but the solution od burning is also wrong because the total pool of snapshot will be  smaller than expected. But in the end such case won't happen so it is better to burn anyway. I just need to filter out the tx.

Reply:
I wouldn't consider burning someone's reward as an error. Burning someone's total balance is an error for sure.